### PR TITLE
Enable colocate_gradients_with_ops=True

### PR DESCRIPTION
**Pull request by kuza55**
By default TensorFlow allocates all gradient matricies on gpu:0, which makes it pretty much impossible to do parallelize a large model.

colocate_gradients_with_ops puts these matricies next to the operations, allowing you to split your model across multiple GPUs. I ran into this issue myself and this fixed it for me.

I think it's also meant to set gradient computations to be done on the device where the operations are stored, but my belief about that comes from https://github.com/tensorflow/tensorflow/issues/2441

I'm not sure why this isn't the default in TF, so I'm not sure if this should be behind a flag or something, but having to make my own patches to keras to do multi-GPU training seems like the wrong answer.

### CODE REVIEW

**Comment by fchollet** [comment]
> I'm not sure why this isn't the default in TF

I think it will be soon. In any case, LGTM.

**Comment by MadhuriNagare** [comment]
@kuza55 and @fchollet  I am using tensorflow 2.0.0 and tf.keras for model definition and training. I am trying to do model parallelism for my multi-input DNN. I am trying to process one input on one GPU. However, when I see the log, every operation is assigned to GPU 0 and when I check nvidia-smi while training, only GPU 0 is being used. 
So I thought, is might be processing forward pass on respective gpus and backward pass on the default GPU 0. So I need to set the flag colocate_gradients_with_ops=True.
However, after going through this thread (and the source code on GitHub), this flag is already true. I am not sure what am I missing.

I tried this test code as well:

[test_code.txt](https://github.com/keras-team/keras/files/5146118/test_code.txt)

I would appreciate any clue about this. Thank you.
